gusto/state.py

```diff
@@ -571,48 +571,56 @@ def _allocate_state(self):
 
 def get_latlon_mesh(mesh):
     coords_orig = mesh.coordinates
-    mesh_dg_fs = VectorFunctionSpace(mesh, "DG", 1)
-    coords_dg = Function(mesh_dg_fs)
-    coords_latlon = Function(mesh_dg_fs)
-    par_loop("""
-for (int i=0; i<3; i++) {
-    for (int j=0; j<3; j++) {
-        dg[i*3 + j] = cg[i*3 + j];
-    }
-}
-""", dx, {'dg': (coords_dg, WRITE),
-          'cg': (coords_orig, READ)})
-
+    coords_fs = coords_orig.function_space()
+
+    if coords_fs.extruded:
+        cell = mesh._base_mesh.ufl_cell().cellname()
+        DG1_hori_elt = FiniteElement("DG", cell, 1, variant="equispaced")
+        DG1_vert_elt = FiniteElement("DG", interval, 1, variant="equispaced")
+        DG1_elt = TensorProductElement(DG1_hori_elt, DG1_vert_elt)
+    else:
+        cell = mesh.ufl_cell().cellname()
+        DG1_elt = FiniteElement("DG", cell, 1, variant="equispaced")
+    vec_DG1 = VectorFunctionSpace(mesh, DG1_elt)
+    coords_dg = Function(vec_DG1).interpolate(coords_orig)
```

Comment:
it'd be good to know why we didn't just do this interpolation in the first place - @dorugeber can you remember? (I think this is code you gave me...)